kb/2025-03-25/cve-2025-1974.md

```diff
@@ -0,0 +1,72 @@
+---
+title: 'CVE-2025-1974: ingress-nginx admission controller RCE escalation'
+description: 'This article describes the mitigation steps for the CVE-2025-1974 vulnerability in Harvester.'
+authors:
+  - name: Ivan Sim
+    title: Principal Software Engineer
+    url: https://github.com/ihcsim
+    image_url: https://github.com/ihcsim.png
+tags: [security, cve]
+hide_table_of_contents: false
+---
+
+:::warning
+
+**CVE-2025-1974** (vector: CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H) has a score of [9.8 (Critical)](https://www.first.org/cvss/calculator/3-1#CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H).
+
+The vulnerability affects specific versions of the RKE2 ingress-nginx controller (v.1.11.4 and earlier, and v1.12.0). All Harvester versions that use this controller (including v1.4.2 and later) are therefore affected.
+
+:::
+
+A security issue was discovered in Kubernetes where under certain conditions, an unauthenticated attacker with access to the pod network can achieve arbitrary code execution in the context of the ingress-nginx controller. This can lead to disclosure of secrets accessible to the controller. (Note that in the default installation, the controller can access all secrets cluster-wide.)
+
+Your Harvester cluster is affected by this vulnerability if:
+
+- the RKE2 ingress-nginx controller is installed, and
+- the pods are running version v1.12.0 or all versions prior to v1.11.4 (included)
+
+You can confirm the version of the RKE2 ingress-nginx pods by running this command on your Harvester cluster:
+
+```sh
+kubectl -n kube-system get po -l"app.kubernetes.io/name=rke2-ingress-nginx" -ojsonpath='{.items[].spec.containers[].image}'
+```
+
+If the command returns one of the affected versions, disable the `rke2-ingress-nginx-admission` validating webhook configuration by performing the following steps:
+
+1. On one of your control plane nodes, create the file `/var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx-config.yaml`.
+
+1. Add the following content to the file:
+
+  ```yaml
+  apiVersion: helm.cattle.io/v1
+  kind: HelmChartConfig
+  metadata:
+    name: rke2-ingress-nginx
+    namespace: kube-system
+  spec:
+    valuesContent: |-
+      controller:
+        admissionWebhooks:
+          enabled: false
+        extraArgs:
+          enable-annotation-validation: true
+  ```
+
+    Harvester automatically applies the change once the content is saved.
+
+RKE2 deletes the `rke2-ingress-nginx-admission` validating webhook configuration and restarts the ingress-nginx pods.
+
+Once your Harvester cluster receives the RKE2 ingress-nginx patch, you can re-install the `rke2-ingress-nginx-admission` validating webhook configuration by removing the `HelmChartConfig` workaround.
```

Comment:
How do I remove the `HelmChartConfig` workaround? I tested this on a random v1.4.0 install I had lying around, and creating `/var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx-config.yaml` as described above results in the rke2-ingress-nginx pod being reconfigured without the validating webhook, but if I then go and delete that manifest file I created, the rke2-ingress-nginx pod doesn't change back. I assume I also need to `kubectl -n kube-system delete helmchartconfig/rke2-ingress-nginx`?

Reply:
users will need to delete the helmchartconfig object

Reply:
I added https://github.com/harvester/harvester/issues/7956 to make sure we handle this during the next upgrade, either via documentation, some codified way, or both. If user already has an existing `HelmChartConfig` object, they need to know they can't just delete it.